MatrixSDK/MXSession.m

```diff
@@ -604,7 +654,55 @@ - (void)resume:(void (^)())resumeDone;
 //        else
         {
             // sync based on API v1 (Legacy)
-            [self streamEventsFromToken:_store.eventStreamToken withLongPoll:NO];
+            if (!eventStreamRequest)
+            {
+                [self streamEventsFromToken:_store.eventStreamToken withLongPoll:NO];
+            }
+        }
+    }
+}
+
+- (void)cancel
```

Comment:
I don't think this 'cancel' is relevant. The existing 'pause' should be used to return in Pause state.